ecs-task-definition-deploy/action.yml

```
@@ -0,0 +1,123 @@
+name: Determine Python and Poetry versions
+description: Determine Python and Poetry versions
+
+inputs:
+  application_id:
+    description: Application ID also the task definition name in this case
+    required: true
+  image_uri:
+    description: The URI of the container image to insert into the ECS task definition
+    required: true
+  deregister_previous:
+    description: Determines whether to deregister previous active task definition
+    required: false
+    default: "false"
+
+  aws_access_key_id:
+    description: AWS access key
+    required: true
+  aws_secret_access_key:
+    description: AWS secret access key
+    required: true
+  aws_region:
+    description: AWS region
+    required: false
+    default: eu-central-1
+
+outputs:
+  previous-task-definition-arn:
+    description: "Python version"
+    value: ${{ steps.get-github-action-task-definition-arn.outputs.task-definition-arn }}
+  latest-task-definition-arn:
+    description: "Python version"
+    value: ${{ steps.deploy-task-definition.outputs.task-definition-arn }}
+
+runs:
+  using: composite
+  steps:
+    - id: configure-aws-credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-access-key-id: ${{ inputs.aws_access_key_id }}
+        aws-secret-access-key: ${{ inputs.aws_secret_access_key }}
+        aws-region: ${{ inputs.aws_region }}
+
+    - id: login-ecr
+      uses: aws-actions/amazon-ecr-login@v2
+
+    - name: Setup Python + Poetry
+      uses: moneymeets/action-setup-python-poetry@master
+      with:
+        working_directory: ${{ github.action_path }}
+        # ToDo: Re-enable cache when https://github.com/actions/setup-python/issues/361 is fixed
+        poetry_cache_enabled: 'false'
+
+    - name: Get latest active task definition created by Pulumi
```

```suggestion
    - name: Get latest active task definition revision created by Pulumi
```
I would still add "revision", just to make it even more clear. Really like this explicit name though, explains exactly what happens here 👍